source/class/qx/data/controller/CheckedList.js

```diff
@@ -0,0 +1,344 @@
+/* ************************************************************************
+
+   qooxdoo - the new era of web development
+
+   http://qooxdoo.org
+
+   Copyright:
+     2021-2021 Zenesis Limited https://www.zenesis.com
+
+   License:
+     MIT: https://opensource.org/licenses/MIT
+     See the LICENSE file in the project's top-level directory for details.
+
+   Authors:
+     * John Spackman (github.com/johnspackman)
+
+************************************************************************ */
+
+/**
+ * Extension of `qx.data.controller.List` which adds support for `qx.ui.form.CheckedList`
+ * and `qx.ui.form.CheckedSelectBox`.
+ *
+ * The principal is thart the underlying `List` controller implementation has a model which
+ * is the complete array of items that can be selected, and that array is used to populate
+ * the UI widget (ie as normal).
+ *
+ * The `checked` psuedo property in this `CheckedList` controller relates to the checked
+ * property of the UI widget.
+ */
+qx.Class.define("qx.data.controller.CheckedList", {
+  extend: qx.data.controller.List,
+  
+  /**
+   * Constructor
+   *
+   * @param model {qx.data.Array?} the model array
+   * @param widget {qx.ui.core.Widget?} the widget target
+   * @param path {String} the path in the model for the caption  
```

Comment:
`path` is optional. Both `widget` and `path` default to null. I think these should be `qx.ui.core.Widget?null` and `String?null` respectively, as in the superclass:

```
   * @param target {qx.ui.core.Widget?null} The widget which should show the
   *   ListItems.
   *
   * @param labelPath {String?null} If the model contains objects, the labelPath
   *   is the path reference to the property in these objects which should be
   *   shown as label.
```